synapse/handlers/room_member.py

```diff
@@ -383,6 +373,20 @@ def _update_membership(
                 if same_sender and same_membership and same_content:
                     defer.returnValue(old_state)
 
+            # we don't allow people to reject invites to the server notice
+            # room, but they can leave it once they are joined.
+            if (
+                old_membership == Membership.INVITE and
+                effective_membership_state == Membership.LEAVE
+            ):
+                is_blocked = yield self._is_server_notice_room(room_id)
+                if is_blocked:
+                    raise SynapseError(
+                        http_client.FORBIDDEN,
+                        "You cannot reject this invite",
+                        errcode=Codes.CANNOT_LEAVE_SERVER_NOTICE_ROOM,
```

Comment:
Is it reasonable to have an errcode of `CANNOT_LEAVE_SERVER_NOTICE_ROOM` when it is about rejecting invites? Or just a mistake?

Reply:
I'm inclined to stick with the error code we have, now that it has started to land in clients, even if it's slightly misleading.